From 719beca22da7196c78202302d57e67d1c0ffc2cb Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:19:41 +0100
Subject: [PATCH] ip6tunnel: print local/remote addresses like iptunnel does

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit 9af72f819e3fa

commit 9af72f819e3fa288fd56e74d14a1253bd49adb9d
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Nov 13 18:08:59 2015 +0100

    ip6tunnel: print local/remote addresses like iptunnel does

    This makes output consistent with iptunnel, also supporting reverse DNS
    lookup for remote address if requested.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 ip/ip6tunnel.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/ip/ip6tunnel.c b/ip/ip6tunnel.c
index f37042e..97c99e2 100644
--- a/ip/ip6tunnel.c
+++ b/ip/ip6tunnel.c
@@ -69,14 +69,17 @@ static void usage(void)
 
 static void print_tunnel(struct ip6_tnl_parm2 *p)
 {
-	char remote[64];
-	char local[64];
-
-	inet_ntop(AF_INET6, &p->raddr, remote, sizeof(remote));
-	inet_ntop(AF_INET6, &p->laddr, local, sizeof(local));
+	char s1[1024];
+	char s2[1024];
 
+	/* Do not use format_host() for local addr,
+	 * symbolic name will not be useful.
+	 */
 	printf("%s: %s/ipv6 remote %s local %s",
-	       p->name, tnl_strproto(p->proto), remote, local);
+	       p->name,
+	       tnl_strproto(p->proto),
+	       format_host(AF_INET6, 16, &p->raddr, s1, sizeof(s1)),
+	       rt_addr_n2a(AF_INET6, 16, &p->laddr, s2, sizeof(s2)));
 	if (p->link) {
 		const char *n = ll_index_to_name(p->link);
 		if (n)
-- 
1.8.3.1

